lib/src/mock_query.dart

```diff
@@ -115,7 +115,7 @@ class MockQuery<T extends Object?> extends FakeQueryWithParent<T> {
       values: values,
       f: (docs, index, exactMatch) => docs.sublist(
             0,
-            index == -1 ? docs.length : index + 1,
```

Comment:
I don't know what the real Firestore behavior is, but in our previous implementation, if the endAt value wasn't found, we would return everything. With your change, now we will return empty. Is that what you are trying to achieve?

Reply:
I will revert this, although the next PR removes this

Reply:
Is it okay if I don't revert this as the next pull request essentially remove this ?  I'm afraid to have to rebase everything after this..

Reply:
Sure, let's keep it simple for you.